src/applications/disability-benefits/686c-674/config/address-schema.js

```diff
@@ -463,18 +489,89 @@ export const addressUISchema = (
         'ui:required': (formData, index) => {
           let militaryBasePath = livesOnMilitaryBasePath;
           let countryNamePath = `${path}.countryName`;
+
           if (typeof index === 'number') {
             militaryBasePath = insertArrayIndex(livesOnMilitaryBasePath, index);
             countryNamePath = insertArrayIndex(countryNamePath, index);
           }
           const livesOnMilitaryBase = get(militaryBasePath, formData);
           const countryName = get(countryNamePath, formData);
+
           return (
             (countryName && countryName === USA.value) ||
             (isMilitaryBaseAddress && livesOnMilitaryBase)
           );
         },
         'ui:title': 'Postal Code',
+        'ui:validations': [
+          (errors, zipCode, formData, _schema, _uiSchema, _index) => {
+            // copied scheme from city ui:validations
+            const livesOnMilitaryBaseHash = {
+              base: get(livesOnMilitaryBasePath, formData),
+              address: get(`address[${MILITARY_BASE_PATH}]`, formData),
+              childAddressInfo: get(
+                `childAddressInfo.address[${MILITARY_BASE_PATH}]`,
+                formData,
+              ),
+            };
+            const validationKeys = [
+              'base',
+              'address',
+              'childAddressInfo',
+              'stepChildAddress',
+              'stepChildAddressInfo',
+            ];
+
+            if (window.location.href.includes('review-and-submit')) {
+              livesOnMilitaryBaseHash.stepchildAddress = (
```

Comment:
Case is different here than in validationKeys

Reply:
great catch!